trio/_tools/gen_exports.py

```diff
@@ -45,17 +52,18 @@ def is_function(node):
     return False
 
 
-def is_public(node):
+def is_public(node: ast.AST) -> TypeGuard[ast.FunctionDef | ast.AsyncFunctionDef]:
     """Check if the AST node has a _public decorator"""
-    if not is_function(node):
-        return False
-    for decorator in node.decorator_list:
-        if isinstance(decorator, ast.Name) and decorator.id == "_public":
-            return True
+    if is_function(node):
+        for decorator in node.decorator_list:
+            if isinstance(decorator, ast.Name) and decorator.id == "_public":
+                return True
     return False
 
 
-def get_public_methods(tree):
+def get_public_methods(
+    tree: ast.AST,
+) -> Generator[ast.FunctionDef | ast.AsyncFunctionDef, None, None]:
```

Comment:
nit: `Iterable` instead of `Generator` means you don't have to specify `None, None`